mobile/native/src/ln_dlc/node.rs

```diff
@@ -200,47 +200,29 @@ impl Node {
-                        let action = decide_subchannel_offer_action(
-                            OffsetDateTime::from_unix_timestamp(
-                                offer.contract_info.get_closest_maturity_date() as i64,
-                            )
-                            .expect("A contract should always have a valid maturity timestamp."),
+                        tracing::info!(
+                            channel_id = offer.temporary_channel_id.to_hex(),
+                            "Automatically accepting dlc channel offer"
                         );
-                        self.process_dlc_channel_offer(offer.temporary_channel_id, action)?;
+                        self.process_dlc_channel_offer(&offer.temporary_channel_id)?;
                     }
                     ChannelMessage::SettleOffer(offer) => {
-                        self.inner
-                            .accept_dlc_channel_collaborative_settlement(&offer.channel_id)
-                            .with_context(|| {
-                                format!(
-                                    "Failed to accept DLC channel close offer for channel {}",
-                                    hex::encode(offer.channel_id)
-                                )
-                            })?;
+                        tracing::info!(
+                            channel_id = offer.channel_id.to_hex(),
+                            "Automatically accepting settle offer"
+                        );
+                        self.process_settle_offer(&offer.channel_id)?;
                     }
                     ChannelMessage::RenewOffer(r) => {
-                        let channel_id_hex = r.channel_id.to_hex();
-
                         tracing::info!(
-                            channel_id = %channel_id_hex,
+                            channel_id = r.channel_id.to_hex(),
                             "Automatically accepting renew offer"
                         );
 
-                        // TODO: This is generally not safe. The coordinator will even send
-                        // `RenewOffer`s in order to roll over, and these will not even be triggered
-                        // by a user action.
-                        let (accept_renew_offer, counterparty_pubkey) =
-                            self.inner.dlc_manager.accept_renew_offer(&r.channel_id)?;
-
-                        self.send_dlc_message(
-                            counterparty_pubkey,
-                            Message::Channel(ChannelMessage::RenewAccept(accept_renew_offer)),
-                        )?;
-
                         let expiry_timestamp = OffsetDateTime::from_unix_timestamp(
                             r.contract_info.get_closest_maturity_date() as i64,
                         )?;
-                        position::handler::handle_channel_renewal_offer(expiry_timestamp)?;
+                        self.process_renew_offer(&r.channel_id, expiry_timestamp)?;
```

Comment:
❓Are these changes meant to be a straight refactor? Might be good to split this out into a separate commit if so. Otherwise I can review this more carefully.

Reply:
I am sorry it became a mixture of a refactoring and adding features. However, I've split the changed policy for dealing with pending offers into a separate commit. I hope that makes it clearer.